### PR TITLE
Setup luarocks deployment workflow

### PR DESCRIPTION
**Pull request by alerque**
So ... this workflow at time of opening this PR is copied from something I've used successfully on other projects (e.g. [cassowary.lua](https://github.com/sile-typesetter/cassowary.lua). That doesn't mean it will be a right fit for this project. It's worth discussing. Here are the questions:

1. Do we want to unify the publishing workflow for all @lunarmodules repos or handle them individually? In other words should some of them be pushed by hand my whoever the maintainer is or should we agree on a workflow that can be codified in CI and automatically publish all of them?

    My suggestion in to unify them all using a dedicated `lunarmodules` user on Luarocks and an API key accessible to all deploy jobs in this org used to publish all projects. This would basically delegate the publishing permissions from whoever holds the relevant Luarocks account(s) to whoever has the correct merge/tag permissions on the relevant GH project.

2. For either this repo or all repos (depending on №1) should the publish action be triggered by posting a new Git tag or a new rockspec file?

    All my programmer instincts suggest having a deploy on tag push would be the *right* way, but my actual experience setting this up for other projects is that the rockspec creation process and the way luarocks.org handles them does not work very well that way. Since several values need to be changed in the rockspec at time of publishing and *those changes* should also be checked via CI, setting up the file first and making sure everything is good, and also allowing force pushes to fix any bloopers that aren't evidenced until you see them on Luarocks is the most pragmatic approach. Thereafter a Git tag seals the deal so that the rockspec can download the tagged sources where the source and rockspec are known to match.

I'll finish this up with some actual code that does what we want once we've settled on answers.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      uses: actions/checkout@v2
+    - name: Setup ‘lua’
+      uses: leafo/gh-actions-lua@v8
+    - name: Setup ‘luarocks’
+      uses: leafo/gh-actions-luarocks@v4
+    - name: Setup dependencies
+      run: |
+        luarocks install dkjson
+    - run: |
+        luarocks upload --force --api-key ${{ secrets.LUAROCKS_APIKEY }} -- ${{ matrix.rockspec }}
```
Comment (daurnimator) [inline .github/workflows/deploy.yml:71]: for bonus points, also upload the .src.rock, and a `.all.rock`

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+name: Deploy
```
Comment (daurnimator) [inline .github/workflows/deploy.yml:1]: I don't think this is a good name

Comment (alerque) [reply .github/workflows/deploy.yml:1]: Do you have one you prefer? “Publish” perhaps?

**Comment by alerque** [comment]
(Heading out for ~7 weeks travel, expect intermittent FOSS participation from me for a while. I'll be back.)

**Comment by alerque** [comment]
Just a quick status upadte, this will be mostly replaced with the streamlined variant I cooked up in [moon-sand](https://github.com/lunarmodules/moon-sand). As soon as the LuaRocks module ownership gets switched to the lunarmodules user I'll move on that. Also we have to decide if I should put it in this repo or in a common one ([.github](https://github.com/lunarmodules/.github) being a likely candidate).

**Pull request merged**